tutorials/plot_spatial.py

```diff
@@ -11,40 +11,237 @@
+
+###################################################################################################
+# 2. Divide position in spatial bin edges and plot
+# ~~~~~~~~
+#
+# Compute x- and y- spatial bin edges and plot spatial grid. 
+#
+
+###################################################################################################
+
+x_edges, y_edges = compute_spatial_bin_edges(position, bins)
+
+# Compute the width of each spatial bin
+x_bins_spatial_width = get_bin_width(x_edges)
+y_bins_spatial_width = get_bin_width(y_edges)
+print(f'The x spatial bins have width = {x_bins_spatial_width}')
+print(f'The y spatial bins have width = {y_bins_spatial_width}')
 
 ###################################################################################################
 
-# Settings
-...
+# Plot: (i) grid of spatial bins, and (ii) grid of spatial bins with tracking on top
+f, (ax1, ax2) = plt.subplots(1, 2)
+# Plot spatial bins in ax1 and ax2
+for i in range(0, max(len(x_edges), len(y_edges))):
+    if i < len(x_edges):
+        l1, = ax1.plot([x_edges[i], x_edges[i]], [y_edges[0], y_edges[-1]], c = 'tab:orange')
+        ax2.plot([x_edges[i], x_edges[i]], [y_edges[0], y_edges[-1]], c = 'tab:orange')
+    if i < len(y_edges):
+        ax1.plot([x_edges[0], x_edges[-1]], [y_edges[i], y_edges[i]], c = 'tab:orange')
+        ax2.plot([x_edges[0], x_edges[-1]], [y_edges[i], y_edges[i]], c = 'tab:orange')
+        
+# Set title and axis labels for left plot (plot (i))
+ax1.set_title('(i) Visualization of spatial bins')
+ax1.set_xlabel('x-position')
+ax1.set_ylabel('y-position')        
+
+# Add position plot to right plot (plot (ii)), for visualization of how the tracking fits in the bins
+track = ax2.plot(x_pos, y_pos, '-x', c = 'tab:gray', markersize = 10)
+ax2.set_title('(ii) Tracking and spatial bins')
+ax2.legend(['spatial bins','Tracking'], loc = 'upper left')
+hl_dict = {handle.get_label(): handle for handle in ax2.get_legend().legendHandles}
+hl_dict['_line1'].set_color('tab:gray')
+hl_dict['_line1'].set_marker('x')
+ax2.set_xlabel('x-position')
+ax2.set_ylabel('y-position')
+
+# Add some padding between subplots
+f.tight_layout(pad = 0.5)
+# Make figure bigger
+f.set_size_inches((20/2.54, 10/2.54))
 
 ###################################################################################################
-# Subtitle
+# 3. Compute spatial bin assignment using spatial bin edges
 # ~~~~~~~~
 #
-# Words, words, words.
+# Compute spatial bin assignment for position using previously computed spatial bin edges.
 #
 
 ###################################################################################################
 
-# Do some code stuff
-print('Hello World!')
+# Now let us check where the 1st 7 point of position using the same x_edges and y_edges
+n_points = 7
+x_bins, y_bins = compute_spatial_bin_assignment(position[:, :n_points], x_edges, y_edges, include_edge=True)
+# We can check they match the positions in plot (ii)
+for i in range (0, n_points):
```

Comment:
there's an extra space after `range` that can be dropped. 

Also in terms of code style, we try to avoid one character variable names (very hard to search for, find & replace etc). For this kind of example, I would tend to use `ind`.